bvm/ballerina-core/src/main/java/org/ballerinalang/BLangProgramRunner.java

```diff
@@ -36,44 +41,71 @@
  */
 public class BLangProgramRunner {
 
-    public static void runService(ProgramFile programFile) {
-        if (!programFile.isServiceEPAvailable()) {
-            throw new BallerinaException("no services found in '" + programFile.getProgramFilePath() + "'");
-        }
-
-        // Get the service package
-        PackageInfo servicesPackage = programFile.getEntryPackage();
-        if (servicesPackage == null) {
-            throw new BallerinaException("no services found in '" + programFile.getProgramFilePath() + "'");
+    public static BValue[] runProgram(ProgramFile programFile, FunctionInfo functionInfo, String[] args) {
+        BValue[] bValArgs;
+        if (functionInfo != null) {
+            bValArgs = extractEntryFuncArgs(functionInfo, args);
+        } else {
+            bValArgs = new BValue[0];
```

Comment:
There can be a scenario, the user has provided args, but a FunctionInfo. In such a case, we are overriding args with empty BValue[], which is incorrect.